lib/APIPlugin.js

```diff
@@ -222,6 +223,20 @@ class APIPlugin {
 							if (key === "__non_webpack_require__" && this.options.module) {
 								/** @type {BuildInfo} */
 								(parser.state.module.buildInfo).needCreateRequire = true;
+								if (
+									this.options.target === "web" ||
+									(Array.isArray(this.options.target) &&
+										this.options.target.includes("web"))
+								) {
+									const warning = new WebpackError(
+										`${PLUGIN_NAME}\nrequire is not allowed in ESM mode`
```

Comment:
Let's make it better - is no allowed in ESM mode with the web target, because we allow require in ESM, we don't allow it only for the `web` target